packages/shortest/src/core/runner/index.ts

```diff
@@ -363,13 +370,40 @@ export class TestRunner {
           }
 
           this.reporter.onTestStart(testCase);
-          const testRun = await this.executeTest(testCase, context);
+          const testRun = new TestRun(testCase);
+          await testRun.initialize();
+          try {
+            testRun.markRunning();
```

Comment:
It needs to be outside becauase `executeTest` calls itself when cached run doesn't work.